src/readii/analyze/correlation.py

```
@@ -80,86 +79,101 @@ def getFeatureCorrelations(vertical_features:pd.DataFrame,
-        msg = "Number of vertical features provided is greater than the number of rows or columns in the correlation matrix."
+        vertical_correlations, horizontal_correlations = getSelfCorrelations(correlation_matrix, vertical_feature_name), getSelfCorrelations(correlation_matrix, horizontal_feature_name)
+
+    except Exception as e:
+        msg = f"Error getting self correlations from correlation matrix: {e}"
         logger.exception(msg)
         raise e
     
-    return correlation_matrix.iloc[0:num_vertical_features, num_vertical_features:]
\ No newline at end of file
+    try:
+        cross_correlations = getCrossCorrelationMatrix(correlation_matrix, vertical_feature_name, horizontal_feature_name)
```

Ruff (0.8.2)

Undefined name `getCrossCorrelationMatrix`

(F821)